linera-rpc/src/simple/server.rs

```diff
@@ -239,15 +239,53 @@ where
                     }
                 }
             }
-            RpcMessage::Certificate(request) => {
+            RpcMessage::TimeoutCertificate(request) => {
+                match self
+                    .server
+                    .state
+                    .handle_timeout_certificate(request.certificate)
+                    .await
+                {
+                    Ok((info, actions)) => {
+                        // Cross-shard requests
+                        self.handle_network_actions(actions);
+                        // Response
+                        Ok(Some(info.into()))
+                    }
+                    Err(error) => {
+                        error!(nickname = self.server.state.nickname(), %error, "Failed to handle certificate");
+                        Err(error.into())
+                    }
+                }
```

Comment:
This could be deduplicated. (Detriplicated?)

Reply:
No, it was just a suggestion. If it's uglier let's not do it!

Reply:
They're not _exactly_ the same due to the notification channel for the `confirmed certificates` case.